### PR TITLE
Allow setting isDone in editTask command

### PR DESCRIPTION
**Pull request by yusiangeng**
resolves https://github.com/AY2223S1-CS2103T-F12-2/tp/issues/56

## Changes

`editTask` command now supports argument `do/`

Usage:
```
# To mark as complete:
editTask 1 do/t
editTask 1 do/true

# To mark as incomplete:
editTask 1 do/f
editTask 1 do/false
```

## Demo

https://user-images.githubusercontent.com/96862218/196477090-b6b00ee8-fa57-4916-b27a-7f434fa808ac.mov

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java [hunk 1]
```diff
@@ -28,15 +28,15 @@ public class AddTaskCommand extends Command {
             + "[" + PREFIX_DESCRIPTION + "DESCRIPTION] "
             + "[" + PREFIX_PRIORITY + "PRIORITY (low/medium/high)] "
             + "[" + PREFIX_CATEGORY + "CATEGORY (database/frontend/backend/uiux/presentation/others)] "
-            + "[" + PREFIX_DEADLINE + "DEADLINE] "
-            + "[" + PREFIX_PERSON + "PERSON Email Address (must be a valid email)]\n"
+            + "[" + PREFIX_DEADLINE + "DEADLINE (YYYY-MM-DD)] "
```
Comment (Kennethsim2000) [inline src/main/java/seedu/address/logic/commands/AddTaskCommand.java:31]: Should we also include that the deadline must be after the current date?

File: src/main/java/seedu/address/logic/commands/EditTaskCommand.java [hunk 2]
```diff
@@ -28,26 +28,25 @@
...
-            + "[" + PREFIX_STATUS + "STATUS]...\n"
+            + "[" + PREFIX_PRIORITY + "PRIORITY (low/medium/high)] "
+            + "[" + PREFIX_CATEGORY + "CATEGORY (database/frontend/backend/uiux/presentation/others)] "
+            + "[" + PREFIX_DEADLINE + "DEADLINE (YYYY-MM-DD)] "
+            + "[" + PREFIX_PERSON + "PERSON EMAIL ADDRESS (must be a valid email)] "
+            + "[" + PREFIX_DONE + "DONE (t/f)]...\n"
             + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_NAME + "Create Initial UI/UX Design"
+            + PREFIX_NAME + "Create Initial UIUX Design"
             + PREFIX_DEADLINE + "2022-02-02";
```
Comment (Kennethsim2000) [inline src/main/java/seedu/address/logic/commands/EditTaskCommand.java:50]: Maybe can change the date for example

File: src/main/java/seedu/address/model/task/TaskCategory.java [hunk 3]
```diff
@@ -12,8 +12,8 @@ public class TaskCategory {
     public static final Integer TASK_CATEGORY_LEVEL_UPPER_BOUND = 5;
     public static final String MESSAGE_CONSTRAINTS =
             "Category name must be one of the following: " + TaskCategoryType.getValidTaskCategories()
-            + " and category level must be >= " + TASK_CATEGORY_LEVEL_LOWER_BOUND.toString()
-            + " and <= " + TASK_CATEGORY_LEVEL_UPPER_BOUND.toString();
+            + " and category level must be >= " + TASK_CATEGORY_LEVEL_LOWER_BOUND
+            + " and <= " + TASK_CATEGORY_LEVEL_UPPER_BOUND;
```
Comment (Kennethsim2000) [inline src/main/java/seedu/address/model/task/TaskCategory.java:16]: Actually not really sure about this isnt this concatenating a string and integer together

**Review by Kennethsim2000 — COMMENTED** [review]
Overally, looks good. Can lmk abt the third review i made abt concatenating an integer and a string. Is that allowed?

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java (hunk 1 above)
Comment (yusiangeng) [inline src/main/java/seedu/address/logic/commands/AddTaskCommand.java:31]:
```suggestion
            + "[" + PREFIX_DEADLINE + "DEADLINE (YYYY-MM-DD, must be after current date)] "
```

File: src/main/java/seedu/address/model/task/TaskCategory.java (hunk 3 above)
Comment (yusiangeng) [reply src/main/java/seedu/address/model/task/TaskCategory.java:16]: intellij was complaining about the `toString()` being redundant, and this works actually

**Pull request merged**